### ISSUE TITLE
"Reads with no misses" graph showing wrong values

### ISSUE THREAD

**Issue by igorribeiroduarte** [scylladb/scylladb#13797]

**Comment by DoronArazii** [comment]
@bhalevy can you please assign someone to provide some insights

**Comment by bhalevy** [comment]
@michoecho please look into this

**Comment by michoecho** [comment]
It's because the workload isn't run from cache, but from memtables.
Memtable reads don't show up in cache stats.

I too was confused by this when I saw it for the first time. Maybe we should have some metrics and panels for memtable reads to make it less confusing.

**Comment by vladzcloudius** [comment]
> It's because the workload isn't run from cache, but from memtables. Memtable reads don't show up in cache stats.
> 
> I too was confused by this when I saw it for the first time. Maybe we should have some metrics and panels for memtable reads to make it less confusing.

@michoecho from the user perspective there are "reads from disk" and "reads from cache". The fact that we internally distinguish between two types of caches used to serve reads is totally irrelevant to the user meaning that reads from memtable are "reads from cache".

**Comment by DoronArazii** [comment]
> It's because the workload isn't run from cache, but from memtables. Memtable reads don't show up in cache stats.
> 
> I too was confused by this when I saw it for the first time. Maybe we should have some metrics and panels for memtable reads to make it less confusing.

@mykaul seems like it's a product/design issue rather than a bug.
What's our plan?

**Comment by mykaul** [comment]
@tzach - thoughts?

**Comment by michoecho** [comment]
I agree with @vladzcloudius. We should add counters for memtable reads and show combined cache+memtable stats in user-facing dashboards.

**Comment by tzach** [comment]
@vladzcloudius point makes sense. 
I suggest presenting memtable and cache in two separate metrics.
Internal implementation has an external effect in this case.

**Comment by DoronArazii** [comment]
@mykaul what's the plan? or the next step for this case?

**Comment by mykaul** [comment]
@tzach - what's the priority for this?

**Comment by michoecho** [comment]
> what's the plan?

1. Implement metrics for memtable reads.
2. Adjust "cache" dashboards in scylla-monitoring to incorporate the memtable metrics.

> or the next step for this case?

Implementing the metrics. I can add this to my queue.

**Comment by DoronArazii** [comment]
Thanks for the explanation Michal.
Removing triage.

**Comment by DoronArazii** [comment]
I guess it won't happen in 5.4
Putting in the backlog until it will have commitment.
